src/averageAges.js

```diff
@@ -55,7 +72,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const peopleWithMother = people.filter(person => people
+    .some(mother => (
+      mother.name === person.mother && (
+        onlyWithSon
+          ? person.sex === 'm'
+          : true
+      )
+    ))
+  );
+
+  const sumOfDifferences = peopleWithMother
+    .reduce((sum, { born: personBorn, mother: motherName }) => {
+      const motherBorn = people
+        .find((person) => person.name === motherName)
+        .born;
+
+      return sum + personBorn - motherBorn;
+    }, 0);
+
+  return Math.round(((sumOfDifferences / peopleWithMother.length) * 100)) / 100;
```

Comment:
Actually this task doesn't require rounding, so you can remove it (but you may leave as it is if you want)